Domain.Tests/EventSourcedAggregateTests.cs

```diff
@@ -404,6 +443,55 @@ public TestCommandPreconditionVerifier(Func<bool> hasBeenApplied)
             {
                 return hasBeenApplied();
             }
+        }
+
+        [Category("Performance")]
+        [Test]
+        public void When_calling_ctor_of_an_aggregate_with_a_large_number_of_source_events_in_non_incrementing_order_then_the_operation_completes_quickly()
+        {
+            var count = 1000000;
+            var largeListOfEvents = Enumerable.Range(1, count).Select(i => new PerfTestAggregate.SimpleEvent { SequenceNumber = i }).ToList();
+            Shuffle(largeListOfEvents, new Random(42));
+
+            var sw = Stopwatch.StartNew();
+            var t = new PerfTestAggregate(Guid.NewGuid(), largeListOfEvents);
+            sw.Stop();
+
+            Console.WriteLine("Elapsed: {0}ms", sw.ElapsedMilliseconds);
+            t.Version.Should().Be(count);
+            t.NumberOfUpdatesExecuted.Should().Be(count);
+            sw.Elapsed.Should().BeLessThan(TimeSpan.FromSeconds(10));
+        }

```

Comment:
2-3 seconds on my machine. As with the other similar test in EventSourceRepositoryTest, I don't want this to be treated as a flaky test if the build server is under loads / people are using a much slower machine. At 1,000,000 events, if another n^2 algorithm is introduced - there is pretty much no chance that this will complete in 10 seconds. That is the primary purpose of this test.